db/init_db.js

```diff
@@ -1,31 +1,193 @@
 // code to build and initialize DB goes here
 const {
-  client
+  client,
   // other db methods 
 } = require('./index');
 
-async function buildTables() {
+const {createUser} = require('./db')
+const {createProduct} = require('./products')
+
+async function dropTables() {
+  client.connect();
+  console.log("Dropping Tables ...")
   try {
-    client.connect();
+    await client.query(`
+      DROP TABLE IF EXISTS order_products;
+      DROP TABLE IF EXISTS orders;
+      DROP TABLE IF EXISTS products;
+      DROP TABLE IF EXISTS users;
+    `);
+  } catch (error) {
+    throw error
+  }
+}
 
-    // drop tables in correct order
+async function buildTables() {
+  console.log("Building Tables ...")
+  try {
+    await client.query(`
+      CREATE TABLE users(
+        id SERIAL PRIMARY KEY,
+        "firstName" VARCHAR(255) NOT NULL,
+        "lastName" VARCHAR(255) NOT NULL,
+        email VARCHAR(255) UNIQUE NOT NULL,
+        imageURL VARCHAR(255) DEFAULT 'imageUrl',
+        username VARCHAR(255) UNIQUE NOT NULL,
+        password VARCHAR(255) UNIQUE NOT NULL,
+        "isAdmin" BOOLEAN DEFAULT false NOT NULL
+      );
+      CREATE TABLE products(
+        id SERIAL PRIMARY KEY,
+        name VARCHAR(255) NOT NULL,
+        description VARCHAR(255) NOT NULL,
+        price VARCHAR(255) NOT NULL,
```

Comment:
Let's use the NUMERIC data type for price, here and below.
https://www.postgresqltutorial.com/postgresql-numeric/
(I don't think we should use "precision" or "scale", just NUMERIC alone)

Reply:
Okay! I think setting precision and scale would be necessary just because a price should have only 2 decimal places so setting scale would set that for us!
I'm going to change it so it accounts for that: price NUMERIC(7,2)

Reply:
Sounds good! I had seen various comments in stackexchange mentioning it's not a good idea to set scale to 2, but I didn't dig into why. We can always change if we run into issues.